# Assignments for "Basic Data Types" 

1. Suppose you invested in Bitcoin at the end of 2017 when Bitcoin gained a lot of value. What would be your money at the end of a week if you had invested \$1000 with an average daily increase of 12\% ? You can solve the problem using Python.

In [1]:
##### Help

# Create a variable capital ($1000)

# Create a variable for daily growth (12%)

# Create a variable for period (7)

# Calculate the final growth rate

# Calculate result

# Print result


In [2]:
# Input values
capital=1000
growth=0.12
period=7

In [3]:
# To calculate gain amount, we will use compound interest formula:
# Amount= Capital * (1 + Growth Rate )**period

In [4]:
amount=capital*(1+growth)**period
print("Result:", amount,"$")
print("Final Growth Rate:", amount-capital,"$")

Result: 2210.6814074060812 $
Final Growth Rate: 1210.6814074060812 $


2. Print the text in quotes with Python. However, you must get the numbers from variables using `.format()` notation. <br> Because the text is long, you might consider writing in two lines:

        `"When we buy bitcoin with 1000 USD at the beginning of the week, we would earn 1210.68 USD at the end of the week, with an average gain of 12\%."`

In [5]:
daily_increase=12
final_growth=amount-capital

In [6]:
print(""" `"When we buy bitcoin with {} USD at the beginning of the week, we would earn {:.2f} USD at the end of the week, \
with an average gain of {}\$.`"  """ .format(capital,final_growth,daily_increase))

 `"When we buy bitcoin with 1000 USD at the beginning of the week, we would earn 1210.68 USD at the end of the week, with an average gain of 12\$.`"  


3. Get the temperature in Fahrenheit from user and write a code to convert it to Celcius. For conversion, you can use this formula: C = (5/9) * (F - 32)

        Enter the temperature in Fahrenheit: 
        user --> 26
        output --> Temperature (C) : -3.33

In [7]:
F=int(input("Enter the temperature in Fahrenheit: "))
C=(5/9)*(F-32)
print("user --> ", F)
print("output --> Temperature (C): {:.2f}".format(C))

Enter the temperature in Fahrenheit:  26


user -->  26
output --> Temperature (C): -3.33


4. Get a three digit number the from user and calculate the sum of the digits in the integer.

        user --> 365
        output --> "The sum of digits in the number is 14

In [8]:
num=input("Please enter a three digit number: ")
type(num)
sum_digits=0
for digit in num:
    sum_digits=sum_digits+int(digit)
print('"The sum of digits in the number is {}."'.format(sum_digits))

Please enter a three digit number:  365


"The sum of digits in the number is 14."


5. Write some code to calculate the hypotenuse of a right angled triangle. Get the side lengths from the user.

        user --> first side lenth : 6
        user --> first side lenth : 8
        output --> "The length of the hypotenuse is 10

In [9]:
import math
side1=float(input("Please enter first side length of a right angled triangle: "))
side2=float(input("Please enter second side length of a right angled triangle: "))
hypotenuse=math.sqrt(side1**2+side2**2)
print('"The length of the hypotenuse is {:.0f}."'.format(hypotenuse))

Please enter first side length of a right angled triangle:  6
Please enter second side length of a right angled triangle:  8


"The length of the hypotenuse is 10."
